res/css/views/rooms/_SearchBar.pcss

```diff
@@ -15,30 +15,43 @@ limitations under the License.
 */
 
 .mx_SearchBar {
-    height: 56px;
+    min-height: 56px; /* for wrapped mx_SearchBar_button making the area tall */
     display: flex;
     align-items: center;
     border-bottom: 1px solid $primary-hairline-color;
 
     .mx_SearchBar_input {
+        --size-button-search: 37px;
+
         /* border: 1px solid $input-border-color; */
         /* font-size: $font-15px; */
         flex: 1 1 0;
         margin-left: 22px;
+
+        /* retain space for the input bar. 7em is a magic number to retain space for the placeholder 'Search…' */
+        min-width: calc(7em + var(--size-button-search));
+
+        input {
+            box-sizing: border-box; /* include padding value into width calculation */
+        }
     }
 
     .mx_SearchBar_searchButton {
+        --size: var(--size-button-search); /* square */
+
         cursor: pointer;
-        width: 37px;
-        height: 37px;
+        width: var(--size);
+        min-width: var(--size);
+        height: var(--size);
         background-color: $accent;
         mask: url("$(res)/img/feather-customised/search-input.svg");
         mask-repeat: no-repeat;
         mask-position: center;
     }
 
     .mx_SearchBar_buttons {
-        display: inherit;
+        display: inherit; /* flex */
+        min-width: 0;
```

Comment:
This min-width declaration is needed as this `mx_SearchBar_buttons` itself is a flex button.

Reply:
I don't understand, I'm afraid. What happens if this is not present?

Reply:
To have the close button displayed even on a *very* narrow timeline.


https://github.com/matrix-org/matrix-react-sdk/assets/3362943/f9ff0d83-85e4-437e-af3e-1f911c7a0dee



I think the buttons should rather be wrapped like on this extreme case, but it is not the theme of this PR (design input should be required too).